bundles/tools.vitruv.testutils/src/tools/vitruv/testutils/views/BasicTestView.xtend

```diff
@@ -0,0 +1,157 @@
+
+/**
+ * A minimal test view that gives access to resources, but does not record any changes.
+ */
+class BasicTestView implements TestView {
+	val Path persistenceDirectory
+	val ResourceSet resourceSet
+	@Accessors
+	val tools.vitruv.testutils.TestUserInteraction userInteraction
+	val UriMode uriMode
+
+	/**
+	 * Creates a test view that will store its persisted resources in the
+	 * provided {@code persistenceDirectory}, and use the provided {@code uriMode}.
+	 */
+	new(Path persistenceDirectory, UriMode uriMode) {
+		this(persistenceDirectory, new tools.vitruv.testutils.TestUserInteraction(), uriMode)
+	}
+
+	/**
+	 * Creates a test view that will store its persisted resources in the
+	 * provided {@code persistenceDirectory}, allow to program interactions through the provided {@code userInteraction},
+	 * and use the provided {@code uriMode}.
+	 */
+	new(
+		Path persistenceDirectory,
+		tools.vitruv.testutils.TestUserInteraction userInteraction,
+		UriMode uriMode
+	) {
+		this(
+			persistenceDirectory,
+			new ResourceSetImpl().withGlobalFactories(),
+			userInteraction,
+			uriMode
+		)
+	}
+
+	/**
+	 * Creates a test view that will store its persisted resources in the provided {@code persistenceDirectory}, access
+	 * resources through the provided {@code resourceSet}, allow to program interactions through the provided
+	 * {@code userInteraction}, and use the provided {@code uriMode}.
+	 */
+	new(
+		Path persistenceDirectory,
+		ResourceSet resourceSet,
+		tools.vitruv.testutils.TestUserInteraction userInteraction,
+		UriMode uriMode
+	) {
+		this.persistenceDirectory = persistenceDirectory
+		this.resourceSet = resourceSet
+		this.userInteraction = userInteraction
+		this.uriMode = uriMode
+	}
+
+	override resourceAt(URI modelUri) {
+		resourceSet.loadOrCreateResource(modelUri)
+	}
+
+	override <T extends EObject> T from(Class<T> clazz, URI modelUri) {
+		val resource = resourceSet.getResource(modelUri, true)
+		return clazz.from(resource)
+	}
+
+	override <T extends Notifier> T record(T notifier, Consumer<T> consumer) {
+		consumer.accept(notifier)
+		return notifier
+	}
+
+	override <T extends Notifier> List<PropagatedChange> propagate(T notifier, Consumer<T> consumer) {
+		val toSave = determineResource(notifier)
+		consumer.accept(notifier)
+		(toSave ?: determineResource(notifier))?.saveOrDelete()
+		return emptyList()
+	}
+
+	override moveTo(Resource resource, Path newViewRelativePath) {
+		resource.URI.absolutePath.moveSafelyTo(persistenceDirectory.resolve(newViewRelativePath))
+		resource.URI = newViewRelativePath.uri
+	}
+
+	override moveTo(Resource resource, URI newUri) {
+		resource.URI.absolutePath.moveSafelyTo(newUri.absolutePath)
+		resource.URI = newUri
+	}
+
+	def private moveSafelyTo(Path source, Path target) {
+		createDirectories(target.parent)
+		move(source, target)
+	}
+
+	def private saveOrDelete(Resource resource) {
+		if (resource.contents.isEmpty) {
+			resource.delete(emptyMap)
```

Comment:
Is there a specific reason for the different handling of the deletion compared to bundles/tools.vitruv.change.propagation/src/tools/vitruv/change/propagation/impl/DefaultChangeRecordingModelRepository.java::delete?

Reply:
I think I am not sure what you mean by "different handling". Could you make that more precise?

Reply:
DefaultChangeRecordingModelRepository catches the IOException and throws an IllegalStateException (and logs the error), BasicTestView does not do any of that.

Reply:
Ah, I see. I have to admit that I have simply taken the code from `ResourceRepositoryImpl`/`ModelInstance` for the `DefaultChangeRecordingModelRepository` without noticing that `BasicTestView` has a similar implementation. I think that catching the exception and throwing a different one does not really make sense here, as an `IOException` is even more specific than the general `IllegalStateException` and logging a error may even not be necessary because the problem will fail anyway (or the one who catches the exception may deal with it properly, which can also be a log entry). So I would propose to remove the additional error logging/retrowing from `DefaultChangeRecordingModelRepository`. Do you agree, @TomWerm?